### ISSUE TITLE
SoftmaxCrossEntropyLoss.evaluate() line 87 throws: ai.djl.engine.EngineException: index 1 is out of bounds for dimension 1 with size 1

### ISSUE THREAD

**Issue by rzorzorzo** [deepjavalibrary/djl#3698]
## Description
simple MLP to train XOR.
input shape: (batch, 2)
output shape: (1)

using the SoftmaxCrossEntropyLoss function randomly throws above exception.

### Expected Behavior
no exception

### Error Message
ai.djl.engine.EngineException: index 1 is out of bounds for dimension 1 with size 1
	at ai.djl.pytorch.jni.PyTorchLibrary.torchGather(Native Method)
	at ai.djl.pytorch.jni.JniUtils.pick(JniUtils.java:581)
	at ai.djl.pytorch.jni.JniUtils.indexAdv(JniUtils.java:417)
	at ai.djl.pytorch.engine.PtNDArrayIndexer.get(PtNDArrayIndexer.java:74)
	at ai.djl.ndarray.NDArray.get(NDArray.java:614)
	at ai.djl.ndarray.NDArray.get(NDArray.java:603)
	at ai.djl.training.loss.SoftmaxCrossEntropyLoss.evaluate(SoftmaxCrossEntropyLoss.java:87)
	at ai.djl.training.EasyTrain.trainSplit(EasyTrain.java:144)
	at ai.djl.training.EasyTrain.trainBatch(EasyTrain.java:116)
	at ai.djl.training.EasyTrain.fit(EasyTrain.java:59)

## Environment Info

win 11 
pytorch 2.5.1+cpu
djl 0.32
pytorch-native-cpu 2.5.1

**Comment by rzorzorzo** [comment]
My bad.
1. For learning XOR best use L2Loss.
2. If using SoftmaxCrossEntropyLoss: SoftmaxCrossEntropyLoss needs to be configured according to your data

in case of the XOR function: there are multiple ways to represent the data:

1. (2) -> (1): eg 0.0f = xor(0.0f, 0.0f)
2. (2) -> (2): eg (0.0f, 0.0f) = xor(0.0f, 0.0f) (inputs are numbers, outputs are labels)
3. (4) -> (2): eg (0.0f, 0.0f) = xor(0.0f, 0.0f, 0.0f, 0.0f) (inputs and outputs are labels)

in case the outputs are not labels as in case 1 above, sparseLabel must be set to false. In the other cases to true.

note that the default value is true.

**Comment by rzorzorzo** [comment]
It would be helpful if the training of the xor function is added to the examples.
